manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml

```diff
@@ -0,0 +1,213 @@
+{{- if .Values.enableArcExtension }}
+{{- if .Values.arc.enableMonitoring }}
+{{- if .Values.linux.enabled}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: arc-{{ template "sscdpa.fullname" . }}-monitoring
+  namespace: {{ .Release.Namespace }}
+{{ include "sscdpa.arc.labels" . | indent 2 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: "arc-{{ template "sscdpa.fullname" . }}"
+  template:
+    metadata:
+{{ include "sscdpa.arc.labels" . | indent 6 }}
+    spec:
+      serviceAccountName: csi-secrets-store-provider-azure
+      hostNetwork: true
+      containers:
+        # Prom MDM Converter
+        - name: prom-mdm-converter
+          image: "{{ .Values.arc.promMdmConverter.image.repository }}:{{ .Values.arc.promMdmConverter.image.tag }}"
+          imagePullPolicy: {{ .Values.arc.promMdmConverter.image.pullPolicy }}
+          env:
+          - name: SERVER_PORT
+            value: "8090"
+          - name: EXTENSION_RESOURCE_ID
+            value: "{{ .Values.Azure.Extension.ResourceId }}"
+          resources:
+{{ toYaml .Values.arc.promMdmConverter.resources | indent 12 }}
+        - name: mdm
+          image: "{{ .Values.arc.mdm.image.repository }}:{{ .Values.arc.mdm.image.tag }}"
+          imagePullPolicy: {{ .Values.arc.mdm.image.pullPolicy }}
+          env: 
+          - name: ROLEINSTANCE
+            valueFrom: 
+              fieldRef: 
+                fieldPath: spec.nodeName
+          - name: CONFIG_OVERRIDES_FILE 
+            value: /tmp/geneva_mdm/mdmconfig.json
+          - name: MDM_INPUT
+            value: statsd_udp,statsd_tcp
+          - name: MDM_LOG_LEVEL 
+            value: "Info"
+          volumeMounts: 
+            - name: mdm-config 
+              mountPath: /tmp/geneva_mdm
+          resources:
+{{ toYaml .Values.arc.mdm.resources | indent 12 }}
+        # MSI Adapter
+        - name: msi-adapter
+          image: "{{ .Values.arc.msiAdapter.image.repository }}:{{ .Values.arc.msiAdapter.image.tag }}"
+          imagePullPolicy: {{ .Values.arc.msiAdapter.image.pullPolicy }}
+          env: 
+          - name: TOKEN_NAMESPACE 
+            value: {{ .Release.Namespace }}
+          - name: EXTENSION_ARMID
+            value: {{ .Values.Azure.Extension.ResourceId }}
+          - name: EXTENSION_NAME 
+            value: {{ .Values.Azure.Extension.Name }}
+          - name: CLUSTER_TYPE 
+            value: ConnectedClusters 
+          - name: CLUSTER_IDENTITY 
+            value: "false" 
+          - name: MANAGED_IDENTITY_AUTH 
+            value: "true" 
+          - name: TEST_MODE 
+            value: "false"
+          resources:
+{{ toYaml .Values.arc.msiAdapter.resources | indent 12 }}
+          securityContext: 
+            capabilities: 
```

Comment:
can we drop all other capabilities as it only requires `NET_ADMIN`?

```yaml
          capabilities:
            drop:
            - ALL
            add:
            - NET_ADMIN
```

Reply:
It'll be good to drop capabilities for other containers too and only add the required ones as they are running on host network

Reply:
removed host network. Also dropping all capabilities results in error for msi-adapter and telegraf

Reply:
If we're not dropping other capabilities, is there a reason for explicitly saying we need `NET_ADMIN`? All the capabilities should be available by default.

Reply:
Without it, in msi-adapter it's not allowing to modify iptable rules.

`error modifying iptable rules: error adding rules to custom chain: running [/sbin/iptables -t nat -N aad-metadata --wait]: exit status 3: iptables v1.8.7 (legacy): can't initialize iptables table nat': Permission denied`

Reply:
thanks for validating!